redis/src/cluster_async/mod.rs

```diff
@@ -499,52 +529,209 @@ where
         Ok(slot_map)
     }
 
-    fn get_connection(&mut self, route: &Route) -> (String, ConnectionFuture<C>) {
-        if let Some((_, node_addrs)) = self.slots.range(&route.slot()..).next() {
-            let addr = node_addrs.slot_addr(route.slot_addr()).to_string();
-            if let Some(conn) = self.connections.get(&addr) {
-                return (addr, conn.clone());
-            }
+    async fn get_connection_from_parts(
+        addr: String,
+        connections_and_metadata: Arc<ConnectionsAndMetadata<C>>,
+    ) -> (String, ConnectionFuture<C>) {
+        if let Some(conn) = connections_and_metadata.connections.lock().await.get(&addr) {
+            return (addr, conn.clone());
+        }
 
-            // Create new connection.
-            //
-            let (_, random_conn) = get_random_connection(&self.connections, None); // TODO Only do this lookup if the first check fails
-            let connection_future = {
-                let addr = addr.clone();
-                let params = self.cluster_params.clone();
-                async move {
-                    match connect_and_check(&addr, params).await {
-                        Ok(conn) => conn,
-                        Err(_) => random_conn.await,
-                    }
+        // Create new connection.
+        //
+        let (_, random_conn) =
+            get_random_connection(&connections_and_metadata.connections, None).await; // TODO Only do this lookup if the first check fails
```

Comment:
Not sure what this todo is for?

Reply:
kept the existing TODO: https://github.com/redis-rs/redis-rs/blob/33b6fede64ff2d298661fef1d907ebdbc6b0d0a2/redis/src/cluster_async/mod.rs#L508